src/Microsoft.Data.Analysis/PrimitiveDataFrameColumn.BinaryOperations.cs

```diff
@@ -1445,6 +1493,7 @@ internal DataFrameColumn AndImplementation<U>(PrimitiveDataFrameColumn<U> column
                 case Type uintType when uintType == typeof(uint):
                 case Type ulongType when ulongType == typeof(ulong):
                 case Type ushortType when ushortType == typeof(ushort):
+                case Type DateTimeType when DateTimeType == typeof(DateTime):
```

Comment:
what is really this code for? I mean what happen if we delete this code?

Reply:
This comment apply to other similar code in this file.

Reply:
All of this is auto-generated from the accompanying [.tt file](https://github.com/dotnet/machinelearning/blob/main/src/Microsoft.Data.Analysis/PrimitiveDataFrameColumn.BinaryOperations.tt).  All of these single line changes are generated off the additional DateTime type added [here](https://github.com/dotnet/machinelearning/pull/6302/files#diff-0772aec3d7a9a5b96abd13ab7a176f48923aef449e251d16ada0260ca2f2a362).  I went with a "don't rock the boat" approach for this and just continued with the existing auto generated patterns. 

Nothing would happen if we deleted this code, but deleting it requires going through the .tt file and finding all the correct places these pointless cases are generated, removing it, and ensuring it has no ill effects. I didn't consider it high priority but could be swayed if you believe it is.

Reply:
I guess changing the order of the types would change the order of these cases and potentially break the experience. I guess that would a decent reason to remove these in favor of the default case.

Reply:
I am not getting this. what order change can affect the behavior? it is a switch statement and testing the type against every case. you cannot have 2 cases equal same type. this whole switch statement can be written as simple if statement.

Reply:
By the way, I am not asking to change it as part of this PR. but will be good if you can track cleaning up tt template files with some issue.